Explorer/Assets/DCL/MapRenderer/ConsumerUtils/MapRenderImage.cs

```diff
@@ -100,48 +104,79 @@ public void OnPointerClick(PointerEventData eventData)
         {
             Profiler.BeginSample(POINTER_CLICK_SAMPLE_NAME);
 
-            if (isActive && !dragging && TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out _))
+            //Process different click types if normal parcel or a map pin
+            if (isActive && !dragging && TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out _, out IPinMarker pinMarker))
                 ParcelClicked?.Invoke(new ParcelClickData
                 {
                     Parcel = parcel,
                     WorldPosition = GetParcelWorldPosition(parcel),
+                    PinMarker = pinMarker
                 });
 
             Profiler.EndSample();
         }
 
         private bool dragging => dragBehavior is { dragging: true };
+        private Vector2Int previousParcel;
 
         private Vector2 GetParcelWorldPosition(Vector2Int parcel)
         {
             Vector2 normalizedDiscretePosition = interactivityController.GetNormalizedPosition(parcel);
             return rectTransform.TransformPoint(rectTransform.rect.size * (normalizedDiscretePosition - rectTransform.pivot));
         }
 
+        private IPinMarker previousMarker;
+
         private void ProcessHover(PointerEventData eventData)
         {
-            if (TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out Vector3 worldPosition))
+            //as in the click process here the hover of different types of underlying element, either map pin or regular parcel
+            //there will be an additional struct as out with the data of the map pin if present, based on that avoid rendering
+            //the parcel highlight and instead render the map pin highlight
+            if (TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out Vector3 worldPosition, out IPinMarker pinMarker))
             {
-                if (highlightEnabled)
-                    interactivityController.HighlightParcel(parcel);
-
-                Hovered?.Invoke(worldPosition);
-                HoveredParcel?.Invoke(parcel);
+                if (highlightEnabled && previousParcel != parcel)
+                {
+                    if (previousMarker != null)
+                    {
+                        previousMarker.AnimateOut();
+                        previousMarker = null;
+                    }
+
+                    previousParcel = parcel;
+                    if (pinMarker == null)
+                    {
+                        interactivityController.HighlightParcel(parcel);
+                    }
+                    else
+                    {
+                        previousMarker = pinMarker;
+                        pinMarker.AnimateIn();
+                        interactivityController.RemoveHighlight();
+                    }

```

Comment:
I think this if should end here after the else? looking at the old code the:
```
                Hovered?.Invoke(worldPosition);
                HoveredParcel?.Invoke(parcel);
```
would be called regardless if highlightEnabled was true or not

Reply:
But I don't think we want to call the Hovered if highlightEnabled is false, also the current usage is related to playing audio clips on hover, that's why I moved them inside

Reply:
then cant a `if (!isHighlighted) return;` be added before the    `if (TryGetParcelUnderPointer...)`? because the next if also checks if is highlighted, so basically both paths need isHighlighted to be true to do anything, I mean, this if's

```
   if (TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out Vector3 worldPosition, out IPinMarker pinMarker))
            {
                if (highlightEnabled && previousParcel != parcel)
```
and this one afterwards:
```
            else if (highlightEnabled)
                interactivityController.RemoveHighlight();
```

so basically, if none of these 2 paths are taken if isHighlighted is false, then we can just return at the top and dont check both things and avoid for example doing the TryGetParcelUnderPointer if isHighlighted is false, right?

Reply:
So it would look something like:
```
if (!highlightEnabled) return;

if (TryGetParcelUnderPointer(...) {
                if (previousParcel != parcel){
                ...}
}
else { 
interactivityController.RemoveHighlight(); 
}
```
wdyt?